### DISCUSSION TITLE
Disabling streaming

### DISCUSSION THREAD

**Discussion by ddzina** [open-webui/open-webui#4065] [Q&A] [answered]
Hello! Can somebody help me to disable default behavior of streaming? I've seen in 0.3.7 release notes that it is possible to set stream to false and use default chat completion (model response appears in the UI only when it generated completely) but I didn't find any documentation or discussion on this. Is there environmental variable or config field to set stream to false?

**Comment by elliottmatt** [comment]
On the chat's "Advanced Settings" there is an option to turn "Stream Chat Response" to "Off".

![image](https://github.com/user-attachments/assets/4b304fc9-91b6-48a7-80b4-5ed1127c1850)

**Reply by recursionbane** [reply]
How can this be disabled for all chat sessions using this model?

**Comment by elliottmatt** [comment, marked as answer]
You can make a new model with custom advanced parameters.

![image](https://github.com/user-attachments/assets/e6f454c5-b457-442d-b935-d19191e62b12)

**Comment by bgeneto** [comment]
Can I disable stream programmatically via Pipe function? How?

**Comment by congzhangzh** [comment]
can I disable it globally?

**Reply by tristan-k** [reply]
So is it possible to disable 'Stream Chat Response' globally?

**Comment by Git2tony** [comment]
For some reason this seems to be default on and it doesn't seem to matter if you switch it off? It leaves my system unworkable since it constantly unloads the Model? (I'm using it with Ollama in WSL2 and have this in my log: 581Z level=WARN source=types.go:514 msg="invalid option provided" option=stream_response)

**Reply by Git2tony** [reply]
(I also disabled the auto-complete), I experience some weird stuff also, like sometimes it answers twice, I get something that looks like a notification in the upper right corner with an answer/remark, and then it is still waiting for answering in the normal chat, which it then also does in the end. It all is dead-slow.

**Comment by HenryXiaoYang** [comment]
Hello, I just want to ask how to change the Stream Response setting globally for all model and all users? As for now, you need to configure the setting for every new chat.

**Comment by zalbiraw** [comment]
I just submitted a PR and have a working image in anyone wants to use that

https://github.com/open-webui/open-webui/pull/18764/files
https://hub.docker.com/r/zalbiraw/open-webui/tags

**Reply by akraines** [reply]
i just made the same exact chagne before I found that you alredy submitted a pr...
https://github.com/open-webui/open-webui/pull/20589
